Client/Frontend/Contextual Hint/ContextualHintViewModel.swift

```diff
@@ -14,78 +14,47 @@ enum CFRTelemetryEvent {
-    typealias CFRStrings = String.ContextualHints
+class ContextualHintViewModel: ContextualHintPrefsKeysProvider {
     typealias CFRPrefsKeys = PrefsKeys.ContextualHints
+    typealias CFRStrings = String.ContextualHints
 
     // MARK: - Properties
     var hintType: ContextualHintViewType
     var timer: Timer?
     var presentFromTimer: (() -> Void)?
     private var profile: Profile
     private var hasSentTelemetryEvent = false
-
     var arrowDirection = UIPopoverArrowDirection.down
-    private var hasAlreadyBeenPresented: Bool {
-        guard let contextualHintData = profile.prefs.boolForKey(prefsKey) else { return false }
-
-        return contextualHintData
-    }
-
-    // Prevent JumpBackIn CFR from being presented if the onboarding
-    // CFR has not yet been presented. On iPad we don't present the onboarding CFR
-    private var canJumpBackInBePresented: Bool {
-        guard UIDevice.current.userInterfaceIdiom != .pad else { return true }
-
-        if let hasShownOnboardingCFR = profile.prefs.boolForKey(CFRPrefsKeys.ToolbarOnboardingKey.rawValue),
-           hasShownOnboardingCFR {
-            return true
-        }
-
-        return false
-    }
-
-    // Do not present contextual hint in landscape on iPhone
-    private var isDeviceHintReady: Bool {
-        !UIWindow.isLandscape || UIDevice.current.userInterfaceIdiom == .pad
-    }
-
-    private var prefsKey: String {
-        switch hintType {
-        case .inactiveTabs: return CFRPrefsKeys.InactiveTabsKey.rawValue
-        case .jumpBackIn: return CFRPrefsKeys.JumpBackinKey.rawValue
-        case .toolbarLocation: return CFRPrefsKeys.ToolbarOnboardingKey.rawValue
-        }
-    }
 
     // MARK: - Initializers
+
     init(forHintType hintType: ContextualHintViewType, with profile: Profile) {
         self.hintType = hintType
         self.profile = profile
     }
 
     // MARK: - Interface
-    func shouldPresentContextualHint() -> Bool {
-        guard isDeviceHintReady else { return false }
-
-        switch hintType {
-        case .jumpBackIn:
-            return canJumpBackInBePresented && !hasAlreadyBeenPresented
 
-        case .toolbarLocation:
-            return SearchBarSettingsViewModel.isEnabled && !hasAlreadyBeenPresented
+    func shouldPresentContextualHint() -> Bool {
+        let contextualHintPresLogic = ContextualHintEligibilityUtility(with: profile)
 
-        default:
-            return !hasAlreadyBeenPresented
-        }
+        return contextualHintPresLogic.canPresent(hintType)
     }
 
     func markContextualHintPresented() {
-        profile.prefs.setBool(true, forKey: prefsKey)
+        switch hintType {
+        // If JumpBackInSyncedTab CFR was shown, don't present JumpBackIn CFR (both convey similar info)
+        case .jumpBackInSyncedTab:
+            profile.prefs.setBool(true, forKey: CFRPrefsKeys.JumpBackInSyncedTabKey.rawValue)
+            profile.prefs.setBool(true, forKey: CFRPrefsKeys.JumpBackinKey.rawValue)
```

Comment:
```suggestion
        // If either JumpBackIN CFR was shown, don't present the other as they both convey similar info
        case .jumpBackInSyncedTab, .jumpBackIn:
            profile.prefs.setBool(true, forKey: CFRPrefsKeys.JumpBackInSyncedTabKey.rawValue)
            profile.prefs.setBool(true, forKey: CFRPrefsKeys.JumpBackinKey.rawValue)
```

I thought if we showed either CFR, we would mark both as shown. IF not' don't mind this comment.

Reply:
SyncedTab expires JBI, but JBI doesn't expire SyncedTab.